### PR TITLE
azure-ai-agents LRO poller util fix for custom end states

### PR DESCRIPTION
**Pull request by jpalvarezl**
This pull request introduces a new internal utility class to centralize and standardize polling behavior for Memory Stores long-running operations (LROs) in the Azure AI Agents SDK. It ensures that required headers are consistently injected and custom terminal states are correctly mapped, while also simplifying and improving the test code for reliability and maintainability.

**Key changes:**

### Polling infrastructure improvements

* Added a new package-private utility class `AgentsServicePollUtils` that injects the required `Foundry-Features` HTTP header into polling requests and remaps custom LRO terminal states ("completed", "superseded") to standard `LongRunningOperationStatus` values. This ensures consistent behavior across all polling strategies.
* Updated `OperationLocationPollingStrategy` and `SyncOperationLocationPollingStrategy` to delegate header injection and status remapping to `AgentsServicePollUtils`, reducing code duplication and improving maintainability. [[1]](diffhunk://#diff-0e26a9483bf2cfa6555777be9adde043987ff812c4dc4910ee99a5bf44c2538dL59-R60) [[2]](diffhunk://#diff-0e26a9483bf2cfa6555777be9adde043987ff812c4dc4910ee99a5bf44c2538dR111-R123) [[3]](diffhunk://#diff-c034404db251109231505ad69e275ab4569e280581ef4dcd28b4a80c4222300fL60-R61) [[4]](diffhunk://#diff-c034404db251109231505ad69e275ab4569e280581ef4dcd28b4a80c4222300fR108-R120)

### Testing enhancements

* Added a new test class `AgentsServicePollUtilsTest` to verify correct header injection and status remapping logic, including edge cases for custom and standard statuses.
* Refactored `MemoryStoresAsyncTests` and `MemoryStoresTests` to remove custom status handling in favor of standard `.isComplete()` checks and added timeouts to prevent hanging tests, improving test reliability and clarity. [[1]](diffhunk://#diff-7a08836956d5000b978abe37ef1ad0b6f3fdf0ff5cae714c1471bc5d81e66680L15-R28) [[2]](diffhunk://#diff-7a08836956d5000b978abe37ef1ad0b6f3fdf0ff5cae714c1471bc5d81e66680L38-L43) [[3]](diffhunk://#diff-7a08836956d5000b978abe37ef1ad0b6f3fdf0ff5cae714c1471bc5d81e66680L285-R284) [[4]](diffhunk://#diff-da57e5ed6bfb67903091b6268fb38503632a5a230c3cd0f3444fc0f66b11b231R11-R19) [[5]](diffhunk://#diff-da57e5ed6bfb67903091b6268fb38503632a5a230c3cd0f3444fc0f66b11b231L26-R32) [[6]](diffhunk://#diff-da57e5ed6bfb67903091b6268fb38503632a5a230c3cd0f3444fc0f66b11b231L129-R136) [[7]](diffhunk://#diff-da57e5ed6bfb67903091b6268fb38503632a5a230c3cd0f3444fc0f66b11b231L237-R240) [[8]](diffhunk://#diff-da57e5ed6bfb67903091b6268fb38503632a5a230c3cd0f3444fc0f66b11b231L305-R305)

### Asset tracking

* Updated the assets manifest to reference the new build/tag for the Java SDK.

### CODE REVIEW

File: sdk/ai/azure-ai-agents/src/main/java/com/azure/ai/agents/implementation/SyncOperationLocationPollingStrategy.java [hunk 1]
```diff
@@ -57,7 +57,8 @@ public SyncOperationLocationPollingStrategy(PollingStrategyOptions pollingStrate
      * @throws NullPointerException if {@code pollingStrategyOptions} is null.
      */
     public SyncOperationLocationPollingStrategy(PollingStrategyOptions pollingStrategyOptions, String propertyName) {
-        super(PollingUtils.OPERATION_LOCATION_HEADER, pollingStrategyOptions);
+        super(PollingUtils.OPERATION_LOCATION_HEADER,
```
Comment (samvaity) [inline sdk/ai/azure-ai-agents/src/main/java/com/azure/ai/agents/implementation/SyncOperationLocationPollingStrategy.java:60]: Are the changes to this file manually added? In that case, won't they be lost on regeneration?

Comment (jpalvarezl) [reply sdk/ai/azure-ai-agents/src/main/java/com/azure/ai/agents/implementation/SyncOperationLocationPollingStrategy.java:60]: They were actually. This puts us in a bit of a bind because as it currently stands, the LROs are just blocking indefinitely waiting for an end state with a name that never happens. And also the preview headers seem to be necessary on each polling request.

**Comment by weidongxu-microsoft** [comment]
The `polling` option does not work. There is some priority in the polling handling and the option would be overridden by the LRO metadata.

Instead, I made the change into Customization class. This should also work.
https://github.com/Azure/azure-sdk-for-java/pull/48209, see last 2 commits.

File: sdk/ai/azure-ai-agents/src/main/java/com/azure/ai/agents/implementation/SyncOperationLocationPollingStrategy.java (hunk 1 above)
Comment (jpalvarezl) [reply sdk/ai/azure-ai-agents/src/main/java/com/azure/ai/agents/implementation/SyncOperationLocationPollingStrategy.java:60]: I addressed this using the suggestion from Weidong here: https://github.com/Azure/azure-sdk-for-java/pull/48157#issuecomment-3992784596
